deploy/operator/upgrade/upgrade.sh

```diff
@@ -1,20 +1,25 @@
 #!/bin/bash
+cd $(pwd)
++source ./common.sh
++source ./utils.sh
 
 echo "************ baremetalds assisted operator upgrade command ************"
-export INDEX_IMAGE="${INDEX_IMAGE:-quay.io/ocpmetal/assisted-service-index:latest}"
-export CATALOG_SOURCE_NAME="${CATALOG_SOURCE_NAME:-community-operators}"
 export CHANNEL="${CHANNEL_UPGRADE_OVERRIDE:-alpha}"
 
-echo "## catalogsource spec.image before upgrade"
-oc get catalogsource assisted-service-operator-catalog -n openshift-marketplace -o=jsonpath='{.spec.image}{"\n"}'
+echo "## subscription spec.channel before upgrade"
+oc get subscription assisted-service-operator -n assisted-installer -o=jsonpath='{.spec.channel}{"\n"}'
 
-# Upgrade the operator
-oc patch catalogsource assisted-service-operator-catalog -n openshift-marketplace --type merge -p "{\"spec\":{\"image\":\"$INDEX_IMAGE\"}}"
-# wait for olm to update the operator and pods to be recreated
-sleep 15m
+# Upgrade the operator via OLM
+oc patch subscription assisted-service-operator -n assisted-installer --type merge -p "{\"spec\":{\"channel\":\"$CHANNEL\"}}"
 
-echo "## catalogsource spec.image after upgrade"
-oc get catalogsource assisted-service-operator-catalog -n openshift-marketplace -o=jsonpath='{.spec.image}{"\n"}'
+# wait for OLM to update the operator and pods to be recreated
+wait_for_condition "pod" "Ready" "22m" "${ASSISTED_NAMESPACE}"  "app=assisted-image-service"
+wait_for_condition "pod" "Ready" "22m" "${ASSISTED_NAMESPACE}"  "app=assisted-image-service"
```

Comment:
Did you mean to wait for both operands?

```suggestion
wait_for_condition "pod" "Ready" "22m" "${ASSISTED_NAMESPACE}"  "app=assisted-service"
wait_for_condition "pod" "Ready" "22m" "${ASSISTED_NAMESPACE}"  "app=assisted-image-service"
```

Reply:
Yes, What I observed is that after updating the subscription channel, when the operator upgrades, at that time `assisted-service` pod is already in `Ready` state from the previous version but we are sure that the `assisted-image-service` is only created after the upgrade. After the upgrade, `assisted-service` pod is also recreated and so I thought of first checking `assisted-image-service` and then `assisted-service`.